packages/indexer/src/indexer.gen.ts

```diff
@@ -1287,44 +1287,89 @@ export class Indexer implements Indexer {
     )
   }
 
-  subscribeReceipts = (args: SubscribeReceiptsArgs, options: WebrpcStreamOptions<SubscribeReceiptsReturn>): Promise<void> => {
+  subscribeReceipts = (
+    args: SubscribeReceiptsArgs,
+    options: WebrpcStreamOptions<SubscribeReceiptsReturn>
+  ): WebrpcStreamController => {
+    const abortController = new AbortController()
+    const abortSignal = abortController.signal
+
+    if (options.signal) {
+      abortSignal.addEventListener('abort', () => abortController.abort(options.signal?.reason), {
+        signal: options.signal
+      })
+    }
+
     const _fetch = () =>
-      this.fetch(this.url('SubscribeReceipts'), createHTTPRequest(args, options.headers, options.signal)).then(
+      this.fetch(this.url('SubscribeReceipts'), createHTTPRequest(args, options.headers, abortSignal)).then(
         async res => {
           await sseResponse(res, options, _fetch)
         },
         error => {
           options.onError(error, _fetch)
         }
       )
-    return _fetch()
+
+    const resp = _fetch()
+    return {
+      abort: abortController.abort.bind(abortController),
+      closed: resp
+    }
   }
-  subscribeEvents = (args: SubscribeEventsArgs, options: WebrpcStreamOptions<SubscribeEventsReturn>): Promise<void> => {
+  subscribeEvents = (args: SubscribeEventsArgs, options: WebrpcStreamOptions<SubscribeEventsReturn>): WebrpcStreamController => {
+    const abortController = new AbortController()
+    const abortSignal = abortController.signal
+
+    if (options.signal) {
+      abortSignal.addEventListener('abort', () => abortController.abort(options.signal?.reason), {
+        signal: options.signal
+      })
+    }
+
     const _fetch = () =>
-      this.fetch(this.url('SubscribeEvents'), createHTTPRequest(args, options.headers, options.signal)).then(
+      this.fetch(this.url('SubscribeEvents'), createHTTPRequest(args, options.headers, abortSignal)).then(
         async res => {
           await sseResponse(res, options, _fetch)
         },
         error => {
           options.onError(error, _fetch)
         }
       )
-    return _fetch()
+
+    const resp = _fetch()
+    return {
+      abort: abortController.abort.bind(abortController),
+      closed: resp
+    }
   }
   subscribeBalanceUpdates = (
     args: SubscribeBalanceUpdatesArgs,
     options: WebrpcStreamOptions<SubscribeBalanceUpdatesReturn>
-  ): Promise<void> => {
+  ): WebrpcStreamController => {
+    const abortController = new AbortController()
+    const abortSignal = abortController.signal
+
+    if (options.signal) {
+      abortSignal.addEventListener('abort', () => abortController.abort(options.signal?.reason), {
+        signal: options.signal
+      })
+    }
```

Comment:
@xiam few improvements to our webrpc streaming API in TypeScript based on https://github.com/webrpc/webrpc/pull/311. Can you take this for a spin please?